test/codepage_test.dart

```diff
@@ -25,24 +27,52 @@ void main() {
     latinThai,
     latinArabic
   ]) {
-    test('${cp.name} codepage', () {
-      // All ASCII compatible.
-      for (var byte = 0x20; byte < 0x7f; byte++) {
-        expect(cp[byte], byte);
-      }
-      // Maps both directions.
-      for (var byte = 0; byte < 256; byte++) {
-        var char = cp[byte];
-        if (char != 0xFFFD) {
-          var string = String.fromCharCode(char);
-          expect(cp.encode(string), [byte]);
-          expect(cp.decode([byte]), string);
+    group('${cp.name} codepage', () {
+      test('ascii compatible', () {
+        for (var byte = 0x20; byte < 0x7f; byte++) {
+          expect(cp[byte], byte);
         }
-      }
-      expect(() => cp.decode([0xfffd]), throwsA(isA<FormatException>()));
-      // Decode works like operator[].
-      expect(cp.decode(bytes, allowInvalid: true),
-          String.fromCharCodes([for (var i = 0; i < 256; i++) cp[i]]));
+      });
+
+      test('bidirectional mapping', () {
+        // Maps both directions.
+        for (var byte = 0; byte < 256; byte++) {
+          var char = cp[byte];
+          if (char != 0xFFFD) {
+            var string = String.fromCharCode(char);
+            expect(cp.encode(string), [byte]);
+            expect(cp.decode([byte]), string);
+          }
+        }
+      });
+
+      test('decode invalid characters not allowed', () {
+        expect(() => cp.decode([0xfffd]), throwsA(isA<FormatException>()));
+      });
+
+      test('decode invalid characters allowed', () {
+        // Decode works like operator[].
+        expect(cp.decode(bytes, allowInvalid: true),
+            String.fromCharCodes([for (var i = 0; i < 256; i++) cp[i]]));
+      });
+
+      test('chunked conversion', () {
+        late final String decodedString;
```

Comment:
Just initialize it to `""` instead of being `late`. (I don't like `late` :wink: )

Reply:
I don't like mutability ;-) I'd prefer to have `late` and make it `final` rather lose the `final` specifier ;-)